src/library_wasmfs.js

```diff
@@ -64,7 +65,15 @@ mergeInto(LibraryManager.library, {
     },
     createPath: (parent, path, canRead, canWrite) => {
       // Cache file path directory names.
-      wasmFSPreloadedDirs.push({parentPath: parent, childName: path});
+      var parts = path.split('/').reverse()
+      while (parts.length) {
+        var part = parts.pop();
+        if (!part) continue;
+        var current = PATH.join2(parent, part);
+        wasmFSPreloadedDirs.push({parentPath: parent, childName: part});
+        parent = current;
+      }
+      return current;
```

Comment:
why is this loop needed now?

Reply:
Because in my new code in _emscripten_fs_load_embedded_files I'm relying on the existing behaviour of FS.createPath, which is that it can recursively create directories.

The current wasmfs version did not support this, and was arguably broken in this respect. So

Reply:
In the old code method we created a pre-js file that called `createPath` once for each directory component, rather than relying on the `createPath` to create its own parents.

Reply:
I see, thanks - I missed that before.